### PR TITLE
[BigQuery] Refactor data tree handlers and write tests

### PR DESCRIPTION
**Pull request by hwing777**
Refactor data tree to do API calls when expanding nodes, instead of loading everything recursively on app load.
Write tests for the refactored handlers.

### CODE REVIEW

File: jupyterlab_bigquery/jupyterlab_bigquery/tests/handlers_test.py [hunk 1]
```diff
@@ -2,14 +2,17 @@
...
+# TODO: add tests for list_projects, list_datasets, list_tables,
 # list_models, get_dataset_details, get_table_details
 
-class TestList(unittest.TestCase):
 
-  def testList(self):
-    pass
+class TestListTree(unittest.TestCase):
+
+  def dummyTest(self):
```
Comment (cbwilkes) [inline jupyterlab_bigquery/jupyterlab_bigquery/tests/handlers_test.py:13]: Is this file still needed?

File: jupyterlab_bigquery/jupyterlab_bigquery/details_handler/details_handler.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import random
+
+import json
+import datetime
+
+from collections import namedtuple
+from notebook.base.handlers import APIHandler, app_log
+from google.cloud import bigquery
+
+# import google.auth
```
Comment (cbwilkes) [inline jupyterlab_bigquery/jupyterlab_bigquery/details_handler/details_handler.py:18]: Delete comments like this in various locations

File: jupyterlab_bigquery/jupyterlab_bigquery/list_items_handler/list_items_handler.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Lint as: python3
+"""Request handler classes for the extensions."""
```
[the remark this answers is not in the document]
Comment (hwing777) [reply jupyterlab_bigquery/jupyterlab_bigquery/list_items_handler/list_items_handler.py:2]: Separated creating handlers to create_handler.py, and decorators to list_items_handler.py

**Comment by hwing777** [comment]
Currently resolving conflicts with Cynthia's last PR

File: jupyterlab_bigquery/jupyterlab_bigquery/tests/list_items_service_test.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+class TestListTree(unittest.TestCase):
+
+  def testListProjects(self):
+    gcp_project = 'dummy_project1'
+
+    mock_client = Mock()
+    mock_client.project = gcp_project
+    bigquery = BigQueryService.get()
+    bigquery._client = mock_client
```
Comment (cbwilkes) [inline jupyterlab_bigquery/jupyterlab_bigquery/tests/list_items_service_test.py:18]: Update your test cases to pass in the mock client. Similar to:

```
    mock_client = Mock()
    mock_client.project = gcp_project
    bigquery = BigQueryService(mock_client)
```

File: jupyterlab_bigquery/jupyterlab_bigquery/list_items_handler/list_items_handler.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+from jupyterlab_bigquery.list_items_handler.create_handler import _handler
+from jupyterlab_bigquery.list_items_handler.service import BigQueryService
+
+SCOPE = ("https://www.googleapis.com/auth/cloud-platform",)
+
+list_items_handlers = {}
```
Comment (cbwilkes) [inline jupyterlab_bigquery/jupyterlab_bigquery/list_items_handler/list_items_handler.py:9]: If my understanding is correct, this variable should be part of the decorator logic and should be moved to be internal to the decorator file.

That way you wouldn't have to pass in the handlers dict. Let me know if I am not understand what you are trying to accomplish with the decorator functionality.
